apps/browser/qml/pages/PrivacySettingsPage.qml

```diff
@@ -110,8 +154,20 @@ Page {
                         remorse = Remorse.popupAction(page, qsTrId("settings_browser-la-cleared_private_data"),
                                                  function() {
                                                      if (clearHistory.checked) {
-                                                         Settings.clearHistory()
-                                                         Settings.removeAllTabs()
+                                                         switch(historyPeriodIndex) {
+                                                         case 0:
+                                                             Settings.clearHistory(HistoryModel.Day)
```

Comment:
It could be like this
Settings.clearHistory(historyPeriods[historyPeriodIndex])